applications/rag/metadata.yaml

```diff
@@ -72,36 +60,114 @@ spec:
+        description: Name of the Kubernetes Backend Config
+        varType: string
+        defaultValue: frontend-iap-config
+      - name: frontend_k8s_backend_service_name
+        description: Name of the Backend Service
+        varType: string
+        defaultValue: rag-frontend
+      - name: frontend_k8s_backend_service_port
+        description: Name of the Backend Service Port
+        varType: number
+        defaultValue: 8080
+      - name: frontend_k8s_iap_secret_name
+        description: Name for frontend iap secret
+        varType: string
+        defaultValue: frontend-iap-secret
+      - name: frontend_k8s_ingress_name
+        varType: string
+        defaultValue: frontend-ingress
+      - name: frontend_k8s_managed_cert_name
+        description: Name for frontend managed certificate
+        varType: string
+        defaultValue: frontend-managed-cert
+      - name: frontend_members_allowlist
+        description: "For example - user:example@google.com,serviceAccount:serviceAccount@google.com,group:group@google.com,domain:google.com"
+        varType: string
+        defaultValue: ""
       - name: gcs_bucket
-        description: GCS bucket name to store dataset
+        description: Bucket name to store the dataset
         varType: string
         required: true
-      - name: jupyter_service_account
-        description: Google Cloud IAM service account for authenticating with GCP services
+      - name: goog_cm_deployment_name
+        varType: string
+        defaultValue: ""
+      - name: jupyter_add_auth
+        varType: bool
+        defaultValue: false
+      - name: jupyter_client_id
+        description: Client ID used for enabling IAP. If empty client credentials will be autogenerated.
+        varType: string
+        defaultValue: ""
+      - name: jupyter_client_secret
+        description: Client secret used for enabling IAP. If empty client credentials will be autogenerated.
         varType: string
-        defaultValue: jupyter
-      - name: k8s_backend_config_name
-        description: Name of the Backend Config on GCP
+        defaultValue: ""
+      - name: jupyter_domain
+        description: Domain used for SSL certificate. If it's empty, *.nip.io DNS is used.
+        varType: string
+        defaultValue: ""
+      - name: jupyter_k8s_backend_config_name
+        description: Name of the Kubernetes Backend Config
         varType: string
         defaultValue: jupyter-iap-config
-      - name: k8s_backend_service_name
-        description: Name of the Backend Config on GCP, this is defined by Jupyter hub
+      - name: jupyter_k8s_backend_service_name
+        description: Name of the Backend Service
         varType: string
         defaultValue: proxy-public
-      - name: k8s_ingress_name
+      - name: jupyter_k8s_backend_service_port
+        description: Name of the Backend Service Port
+        varType: number
+        defaultValue: 80
+      - name: jupyter_k8s_iap_secret_name
```

Comment:
Do we need these if they're not exposed in the UI?

Reply:
Not exactly but the blueprint generates metadata.yaml from variables.tf.. so it is better to manage here.. so if blueprint is triggered again.. it will respect the settings already configured.